src/QMCWaveFunctions/tests/test_ConstantSPOSet.cpp

```diff
@@ -0,0 +1,193 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2023 QMCPACK developers.
+//
+// File developed by: Raymond Clay, rclay@sandia.gov, Sandia National Laboratories
+//
+// File created by: Raymond Clay, rclay@sandia.gov, Sandia National Laboratories
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#include "catch.hpp"
+#include "Configuration.h"
+#include "QMCWaveFunctions/OrbitalSetTraits.h"
+#include "QMCWaveFunctions/ConstantSPOSet.h"
+
+namespace qmcplusplus
+{
+//Ray:  Figure out how to template me on value type.
+TEST_CASE("ConstantSPOSet", "[wavefunction]")
+{
+  //For now, do a small square case.
+  const int nelec = 2;
+  const int norb  = 2;
+
+  using RealType    = QMCTraits::RealType;
```

Comment:
I'd suggest using src/QMCWaveFunctions/WaveFunctionTypes.hpp to define consistent types.  QMCTraits bakes in the multiexecutable design and is just bad. OrbitalSetTraits obfuscates the handling of precision.

example from `DiracDeterminantBatched`
```
  using WFT           = WaveFunctionTypes<VALUE, FP_VALUE>;
  using Value         = typename WFT::Value;
  using FullPrecValue = typename WFT::FullPrecValue;
  using PsiValue      = typename WFT::PsiValue;
  using LogValue      = typename WFT::LogValue;
  using Grad          = typename WFT::Grad;
  using Hess          = typename WFT::Hess;
  using Real          = typename WFT::Real;
```
Vector types gain no value from the typedefs at all since they can just be written
```
Vector<Value> row0;
Vector<Grad> grow0;
Vector<Value> lrow0l;
```

Reply:
WaveFunctionTypes is the file I need, thank you.  I'll make those changes

Reply:
Done